ipapython/ipaldap.py

```diff
@@ -1166,6 +1210,9 @@ def simple_bind(self, bind_dn, bind_password, server_controls=None,
         """
         Perform simple bind operation.
         """
+        if self.protocol == 'ldap' and not self._start_tls and bind_password:
+            # non-empty bind must use a secure connection
+            raise ValueError('simple_bind over insecure LDAP connection')
```

Comment:
Similar question. Is it better to catch this at construction or at bind? If protocol == ldap and not start_tls then regardless of password this is bad, right?

Reply:
No, it is not necessarily bad. GSSAPI + SASL also provide confidentiality and strong authentication over a plain LDAP connection. IPA uses plain LDAP + GSSAPI to fetch the CA certificate. Only password binds over plain LDAP are bad, because the combination transmits the password in plain text.